frame-metadata/src/lib.rs

```diff
@@ -298,63 +375,91 @@ impl core::fmt::Debug for DefaultByteGetter {
 
 /// Hasher used by storage maps
 #[derive(Clone, PartialEq, Eq, Encode)]
-#[cfg_attr(feature = "std", derive(Decode, Serialize, Debug))]
+#[cfg_attr(feature = "std", derive(Decode, Serialize, Deserialize, Debug))]
 pub enum StorageHasher {
+	/// 128-bit Blake2 hash.
 	Blake2_128,
+	/// 256-bit Blake2 hash.
 	Blake2_256,
+	/// 128-bit Blake2 concatenating multiple hashes.
 	Blake2_128Concat,
+	/// 128-bit XX hash.
 	Twox128,
+	/// 256-bit XX hash.
 	Twox256,
+	/// 64-bit XX hashes concatentation.
 	Twox64Concat,
+	/// Identity hashing (no hashing).
 	Identity,
 }
 
 /// A storage entry type.
 #[derive(Clone, PartialEq, Eq, Encode)]
-#[cfg_attr(feature = "std", derive(Decode, Serialize, Debug))]
+#[cfg_attr(feature = "std", derive(Decode, Serialize, Deserialize, Debug))]
 pub enum StorageEntryType {
+	/// Plain storage entry (just the value).
 	Plain(DecodeDifferentStr),
+	/// A storage map.
 	Map {
+		/// Hasher type for the keys.
 		hasher: StorageHasher,
+		/// Key type.
 		key: DecodeDifferentStr,
+		/// Value type.
 		value: DecodeDifferentStr,
-		// is_linked flag previously, unused now to keep backwards compat
+		///
+		/// NOTE is_linked flag previously, unused now to keep backwards compat
+		/// with SCALE encoding.
 		unused: bool,
 	},
+	/// Storage Double Map.
 	DoubleMap {
+		/// Hasher type for the keys.
 		hasher: StorageHasher,
+		/// First key type.
 		key1: DecodeDifferentStr,
+		/// Second key type.
 		key2: DecodeDifferentStr,
+		/// Value type.
 		value: DecodeDifferentStr,
+		/// Hasher for the second key.
 		key2_hasher: StorageHasher,
 	},
 }
 
 /// A storage entry modifier.
 #[derive(Clone, PartialEq, Eq, Encode)]
-#[cfg_attr(feature = "std", derive(Decode, Serialize, Debug))]
+#[cfg_attr(feature = "std", derive(Decode, Serialize, Deserialize, Debug))]
 pub enum StorageEntryModifier {
```

Comment:
dq: what is a `StorageEntryModifier`?

Reply:
The code is literally just a copy-paste of `frame-metadata` from substrate repo. It defines if a non existent value should end up being `None` or should fall back to `Default` value when read.